core/src/main/java/ar/com/wolox/wolmo/core/presenter/BasePresenter.kt

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright (c) Wolox S.A
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package ar.com.wolox.wolmo.core.presenter
+
+import androidx.core.util.Consumer
+import javax.inject.Inject
+
+/**
+ * Base presenter that provides the view to the specific presenters.
+ */
+@Suppress("DeprecatedCallableAddReplaceWith")
+open class BasePresenter<V> @Inject constructor() {
+
+    /**
+     * Returns the [view] attached this presenter.
+     * **NOTE: ** You should check if the view is attached calling the
+     * method (before calling any method of the view).[BasePresenter.isViewAttached]
+     */
+    var view: V? = null
+        private set(value) {
```

Comment:
I'm pretty sure you don't need to define the setter, just put `private set` and I think it'll be fine.